bin/juttle

```diff
@@ -214,6 +217,10 @@ function perform_run(options) {
             program.deactivate();
         }).finally(function() {
             current_program = undefined;
+
+            if (had_an_error) {
+                return Promise.reject();
```

Comment:
@mstemm I'm not altering the error reporting just making sure the CLI exits with a non zero. I will file a separate issue for the error reporting to be less verbose than the current spewing of JSON as seen above. The question I have here is since @demmer wants the rejection to inclue an `Error` what should the message be ? Something like `Juttle program failed, see above for details`, `Program terminated with errors` ?  I wasn't including an `Error` to begin with so that there wasn't another line indicating there were errors when there's already plenty of logging going on and the CLI would actually exit with a non zero code indicating there was something to look at.